xchange-stream-service-netty/src/main/java/info/bitrich/xchangestream/service/netty/NettyStreamingService.java

```diff
@@ -385,12 +394,15 @@ public void sendMessage(String message) {
   }
 
   public void resubscribeChannels() {
-    for (Entry<String, Subscription> entry : channels.entrySet()) {
-      try {
-        Subscription subscription = entry.getValue();
-        sendMessage(getSubscribeMessage(subscription.channelName, subscription.args));
-      } catch (IOException e) {
-        LOG.error("Failed to reconnect channel: {}", entry.getKey());
+    if (channels.size() > 0) {
```

Comment:
This feels side-effecty. If you really do want to warn when this is called, it shouldn't not warn when the list is empty. If anything, I might expect the opposite case: a warning when resubscribe is called WITHOUT anything to resubscribe to.

Also, the if makes the whole method more indented for no reason: If the list is 0, the for will execute 0 times regardless, so there's no cost savings here, just more complexity of code.

Reply:
right now, a warning is displayed when the socket first connects, before any subscriptions are made, and this is normal behavior.  I dont think we need this warning at all, probably should only be a debug statement, if at all.

Reply:
+1 for getting rid of it. :)